### ISSUE TITLE
IntSlice does not work well with environment variable

### ISSUE THREAD

**Issue by adriantam** [spf13/viper#1611]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/spf13/viper/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/spf13/viper/issues/new/choose) without success.
- [X] I have checked the [troubleshooting guide](https://github.com/spf13/viper/blob/master/TROUBLESHOOTING.md) for my problem, without success.

### Viper Version

1.16.0

### Go Version

1.20.7

### Config Source

Manual set

### Format

Other (specify below)

### Repl.it link

https://replit.com/@adriantam5/ViperIntSliceEnvironmentVariable

### Code reproducing the issue

```go
package main

import (
	"fmt"
	"os"

	"github.com/spf13/viper"
)

func main() {

	viper.SetConfigType("env")

	viper.AutomaticEnv()
	viper.BindEnv("MYSTR")
	viper.BindEnv("MYINT")
	viper.BindEnv("MYSTR_SLICE")
	viper.BindEnv("MYINT_SLICE")

	os.Setenv("MYSTR", "aaa")
	os.Setenv("MYINT", "29")
	os.Setenv("MYSTR_SLICE", "xyz def")
	os.Setenv("MYINT_SLICE", "15 30")

	mystr := viper.GetString("MYSTR")
	fmt.Println("string", mystr)

	myint := viper.GetString("MYINT")
	fmt.Println("integer", myint)

	mystr_slice := viper.GetStringSlice("MYSTR_SLICE")
	fmt.Println("string slice", mystr_slice)

	myint_slice := viper.GetIntSlice("MYINT_SLICE")
	fmt.Println("int slice", myint_slice)
}
```

### Expected Behavior

Expected: myint_slice return []integer{15,30}

### Actual Behavior

Actual: myint_slice returns empty slice

### Steps To Reproduce

See code above

**Comment by sagikazarmark** [comment]
Thank you for opening this issue. Unfortunately the type conversions provided by Viper are partly implemented by a different library (cast).

This particular feature (converting a space delimited string to a slice) is implemented by cast not Viper itself.

Unfortunately, converting a space delimited string to an int slice is much more complex (eg. what do you do with `123 haha`?)

There is a solution that you could use and should work today, but it requires a bit different usage pattern for Viper and is documented here: https://sagikazarmark.hu/blog/decoding-custom-formats-with-viper/

tl;dr using custom decoding hooks you can implement a parser for any kind of format.

Due to the complex nature of the problem, I can't promise this issue will be prioritized in the near future.
